src/app/modules/main/controller.nim

```diff
@@ -229,12 +229,17 @@ proc getActiveSectionId*(self: Controller): string =
   result = self.activeSectionId
 
 proc setActiveSection*(self: Controller, sectionId: string, sectionType: SectionType) =
+  if sectionType == SectionType.Community and
+      not singletonInstance.localAccountSensitiveSettings.getCommunitiesEnabled():
```

Comment:
I am just wondering if this setting shouldn't determine backend behavior as well, that is: to not build/maintain communities data/models when disabled. If that is the case this check would not be needed here.

Reply:
@osmaczko you have right, we shouldn't get here if community is not enabled, so the real reason why does this proc get triggered should be somewhere else.

As I see this is called in case if someone makes a certain community active from the qml side (that should not be the case if communities are disabled, in that case they are not displayed in the app), and there are some more cases like `SIGNAL_COMMUNITY_JOINED`, `SIGNAL_COMMUNITY_CREATED`, `SIGNAL_COMMUNITY_IMPORTED`, `SIGNAL_COMMUNITY_LEFT`. In all these cases we shouldn't emit community related signals if communities are disabled.

Reply:
> The only downside is that if we do that, we would need to force the user to reboot the app for the changes to take effect, because we'd need to load all communities, build the UI, etc.

What if we block all signals from the service? 
Instead of:
`self.events.emit(SIGNAL_COMMUNITY_JOINED, CommunityArgs(community: community))`
we could:
`self.maybeEmit(SIGNAL_COMMUNITY_JOINED, CommunityArgs(community: community))`
where `maybeEmit` is:
```
  proc maybeEmit(self: Service, signal string, args CommunityArgs) =
    if singletonInstance.localAccountSensitiveSettings.getCommunitiesEnabled():
      self.events.emit(signal, args)
```

Please let me know what you think about this solution.